polar-core/src/vm.rs

```diff
@@ -2159,15 +2160,36 @@ impl PolarVirtualMachine {
     ///  - Failure => backtrack
     fn unify(&mut self, left: &Term, right: &Term) -> PolarResult<()> {
         match (left.value(), right.value()) {
-            (Value::Expression(_), _) | (_, Value::Expression(_)) => {
-                return Err(self.type_error(
-                    left,
-                    format!(
-                        "cannot unify expressions directly `{}` = `{}`",
-                        left.to_polar(),
-                        right.to_polar()
-                    ),
-                ));
+            (Value::Expression(op), other) | (other, Value::Expression(op)) => {
+                match op {
+                    // this branch handles dot ops that were rewritten for inclusion
+                    // in a partial, but then queried again after the partial was bound;
+                    // cf. Vm::dot_op_helper(), Vm::bind()
```

Comment:
https://en.wikipedia.org/wiki/Cf.

Reply:
Ok, it would be clearer to replace this with something like


> this branch handles dot ops that were rewritten for inclusion in a partial in Vm::dot_op_helper() but then queried again after the partial was bound by a call to Vm::bind()

For the same reasons that we try to avoid abbreviations. It makes reading the code harder if you don't know what `cf.` is.